// db3.js/src/proto/db3_bill.ts
// @generated by protobuf-ts 2.8.2 with parameter optimize_code_size,long_type_string
// @generated from protobuf file "db3_bill.proto" (package "db3_bill_proto", syntax proto3)
// tslint:disable
//
//
//
// db3_bill.proto
// Copyright (C) 2022 db3.network Author imotai <codego.me@gmail.com>
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
//
import { MessageType } from "@protobuf-ts/runtime";
/**
 * @generated from protobuf message db3_bill_proto.Bill
 */
export interface Bill {
    /**
     * uint256
     *
     * @generated from protobuf field: uint64 gas_fee = 1;
     */
    gasFee: string;
    /**
     * the block id
     *
     * @generated from protobuf field: uint64 block_id = 2;
     */
    blockId: string;
    /**
     * query session or mutation
     *
     * @generated from protobuf field: db3_bill_proto.BillType bill_type = 4;
     */
    billType: BillType;
    /**
     * time on block created
     *
     * @generated from protobuf field: uint64 time = 5;
     */
    time: string;
    /**
     * mutation id or query session id
     *
     * @generated from protobuf field: bytes tx_id = 6;
     */
    txId: Uint8Array;
    /**
     * owner address
     *
     * @generated from protobuf field: bytes owner = 8;
     */
    owner: Uint8Array;
    /**
     * to address
     *
     * @generated from protobuf field: bytes to = 9;
     */
    to: Uint8Array;
}
/**
 * @generated from protobuf message db3_bill_proto.BillQueryRequest
 */
export interface BillQueryRequest {
    /**
     * @generated from protobuf field: uint64 block_height = 1;
     */
    blockHeight: string;
}
/**
 * @generated from protobuf enum db3_bill_proto.BillType
 */
export enum BillType {
    /**
     * @generated from protobuf enum value: BillForMutation = 0;
     */
    BillForMutation = 0,
    /**
     * @generated from protobuf enum value: BillForQuery = 1;
     */
    BillForQuery = 1,
    /**
     * @generated from protobuf enum value: BillForMint = 2;
     */
    BillForMint = 2
}
// @generated message type with reflection information, may provide speed optimized methods
class Bill$Type extends MessageType<Bill> {
    constructor() {
        super("db3_bill_proto.Bill", [
            { no: 1, name: "gas_fee", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 2, name: "block_id", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 4, name: "bill_type", kind: "enum", T: () => ["db3_bill_proto.BillType", BillType] },
            { no: 5, name: "time", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 6, name: "tx_id", kind: "scalar", T: 12 /*ScalarType.BYTES*/ },
            { no: 8, name: "owner", kind: "scalar", T: 12 /*ScalarType.BYTES*/ },
            { no: 9, name: "to", kind: "scalar", T: 12 /*ScalarType.BYTES*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message db3_bill_proto.Bill
 */
export const Bill = new Bill$Type();
// @generated message type with reflection information, may provide speed optimized methods
class BillQueryRequest$Type extends MessageType<BillQueryRequest> {
    constructor() {
        super("db3_bill_proto.BillQueryRequest", [
            { no: 1, name: "block_height", kind: "scalar", T: 4 /*ScalarType.UINT64*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message db3_bill_proto.BillQueryRequest
 */
export const BillQueryRequest = new BillQueryRequest$Type();
