// db3.js/src/proto/db3_message.ts
// @generated by protobuf-ts 2.8.2 with parameter optimize_code_size,long_type_string
// @generated from protobuf file "db3_message.proto" (package "db3_message_proto", syntax proto3)
// tslint:disable
import { MessageType } from "@protobuf-ts/runtime";
/**
 * @generated from protobuf message db3_message_proto.DepositEvent
 */
export interface DepositEvent {
    /**
     * the evm chain id
     *
     * @generated from protobuf field: uint32 chain_id = 1;
     */
    chainId: number;
    /**
     * the evm chain user address
     *
     * @generated from protobuf field: bytes sender = 2;
     */
    sender: Uint8Array;
    /**
     * the deposit amount
     *
     * @generated from protobuf field: uint64 amount = 3;
     */
    amount: string;
    /**
     * the block id of evm chain
     *
     * @generated from protobuf field: uint64 block_id = 4;
     */
    blockId: string;
    /**
     * the transacion id
     *
     * @generated from protobuf field: bytes transaction_id = 5;
     */
    transactionId: Uint8Array;
    /**
     * the signature of transacion
     *
     * @generated from protobuf field: bytes signature = 6;
     */
    signature: Uint8Array;
    /**
     * @generated from protobuf field: bytes tx_signed_hash = 7;
     */
    txSignedHash: Uint8Array;
}
// @generated message type with reflection information, may provide speed optimized methods
class DepositEvent$Type extends MessageType<DepositEvent> {
    constructor() {
        super("db3_message_proto.DepositEvent", [
            { no: 1, name: "chain_id", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 2, name: "sender", kind: "scalar", T: 12 /*ScalarType.BYTES*/ },
            { no: 3, name: "amount", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 4, name: "block_id", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 5, name: "transaction_id", kind: "scalar", T: 12 /*ScalarType.BYTES*/ },
            { no: 6, name: "signature", kind: "scalar", T: 12 /*ScalarType.BYTES*/ },
            { no: 7, name: "tx_signed_hash", kind: "scalar", T: 12 /*ScalarType.BYTES*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message db3_message_proto.DepositEvent
 */
export const DepositEvent = new DepositEvent$Type();
