// db3.js/src/proto/db3_account.ts
// @generated by protobuf-ts 2.8.2 with parameter optimize_code_size,long_type_string
// @generated from protobuf file "db3_account.proto" (package "db3_account_proto", syntax proto3)
// tslint:disable
import { MessageType } from "@protobuf-ts/runtime";
/**
 *
 *
 *
 * @generated from protobuf message db3_account_proto.Account
 */
export interface Account {
    /**
     * @generated from protobuf field: uint64 bills = 1;
     */
    bills: string;
    /**
     * @generated from protobuf field: uint64 credits = 2;
     */
    credits: string;
    /**
     * @generated from protobuf field: uint64 total_storage_in_bytes = 3;
     */
    totalStorageInBytes: string;
    /**
     * @generated from protobuf field: uint64 total_mutation_count = 4;
     */
    totalMutationCount: string;
    /**
     * @generated from protobuf field: uint64 total_session_count = 5;
     */
    totalSessionCount: string;
    /**
     * @generated from protobuf field: uint64 nonce = 6;
     */
    nonce: string;
}
// @generated message type with reflection information, may provide speed optimized methods
class Account$Type extends MessageType<Account> {
    constructor() {
        super("db3_account_proto.Account", [
            { no: 1, name: "bills", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 2, name: "credits", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 3, name: "total_storage_in_bytes", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 4, name: "total_mutation_count", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 5, name: "total_session_count", kind: "scalar", T: 4 /*ScalarType.UINT64*/ },
            { no: 6, name: "nonce", kind: "scalar", T: 4 /*ScalarType.UINT64*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message db3_account_proto.Account
 */
export const Account = new Account$Type();
